Dockerfile.prod

```diff
@@ -0,0 +1,42 @@
+# ----------------------------------------------------------------
+# This Dockerfile uses multiple best practices recommended by Snyk,
+# OWASP, and Docker. In this file the practices are defined, not 
+# explained. For more in depth explaination refer to the
+# security-documentation repository.
+#
+# This file should be used for production purposes. If building
+# an image for development use Dockerfile.dev
+# ----------------------------------------------------------------
+# ----------------------------------------------------------------
+# Build Stage
+# ----------------------------------------------------------------
+FROM node:latest as build
+WORKDIR /usr/src/app
+COPY package.json tsconfig.json /usr/src/app/
+RUN npm install
+
+COPY  ./src /usr/src/app/src
+COPY  ./docs /usr/src/app/docs
+
+# ----------------------------------------------------------------
+# Serve Stage
+# ----------------------------------------------------------------
+# Specifies image and version
+FROM node:lts-alpine@sha256:f07ead757c93bc5e9e79978075217851d45a5d8e5c48eaf823e7f12d9bbc1d3c
+# Process supervisor that forwards signals to children
+RUN apk add dumb-init
+RUN apk --no-cache add --virtual builds-deps build-base python
+# Specifies that packages should run as prod
+WORKDIR /usr/src/app
+# Changes ownership of files from root to node
+COPY --chown=node:node --from=build /usr/src/app /usr/src/app
+RUN npm run build
```

Comment:
Can we actually move the npm run build to the build stage and then only copy over the compiled js. It'll cut down on size and won't expose source code.

Ideally, have two COPY commands, 1. compiled js, 2. node_modules